core/src/main/java/software/amazon/awssdk/retry/conditions/RetryOnErrorCodeCondition.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.retry.conditions;
+
+import java.util.HashSet;
+import java.util.Set;
+import software.amazon.awssdk.AmazonServiceException;
+import software.amazon.awssdk.retry.RetryPolicyContext;
+
+public class RetryOnErrorCodeCondition implements RetryCondition {
+
+    private final Set<String> retryableErrorCodes;
+
+    // TODO: Switch to varargs and Set.of()
+    public RetryOnErrorCodeCondition(Set<String> retryableErrorCodes) {
+        this.retryableErrorCodes = new HashSet<>(retryableErrorCodes);
+    }
```

Comment:
With Java 9 will we need varargs? Set.of(...) isn't much harder, and it makes it clear that there's no duplicates allowed. I don't have a strong opinion, but there are a few other conditions that take Sets in this review, so we should consider moving this TODO to them as well.

Reply:
A little less burden on the customer. Fine either way though.